src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/Server/SqlDataRecord.cs

```diff
@@ -283,22 +196,20 @@ public virtual bool IsDBNull(int ordinal)
         //  ISqlRecord implementation
         public virtual SqlMetaData GetSqlMetaData(int ordinal)
         {
-            EnsureSubclassOverride();
+            ThrowIfInvalidOrdinal(ordinal);
```

Comment:
Could we call this on all APIs accepting ordinal? As I see the internal ValueUtilSmi does not do any significant checks.

Reply:
As I mentioned in the original description there are either direct or indirect calls to `ThrowIfInvalidOrdinal` on every entry point now because `GetMetaData` and `GetSmiMetaData` call `ThrowIfInvalidOrdinal`, so we don't need a direct call each time it'll just cause performance noise, we'll get the exception on any ordinal (unless you see one I missed).